core/src/main/scala/cr/pulsar/Consumer.scala

```diff
@@ -66,27 +70,18 @@ trait Consumer[F[_], E] {
 object Consumer {
 
   case class Message[A](id: MessageId, key: MessageKey, payload: A)
-  case class DecodingFailure(bytes: Array[Byte]) extends NoStackTrace
-
-  sealed trait OnFailure[+F[_]]
```

Comment:
This is not needed anymore?

Reply:
No, with schemas we don't decode on consuming, Pulsar does it for us.

Reply:
Imo, the schema should be an optional functionality in the lib

Reply:
Then we need lots of duplication of code, can't reuse the same pieces, unfortunately.

Reply:
I think we have discussed this on Slack. We already use `Schema.BYTES`, that's the default, a schema is not optional.

Reply:
Yeah, I remembered, that makes sense